<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>798</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="my_label" name="frame">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>25</y>
      <width>750</width>
      <height>450</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QLabel" name="mouse_movement">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>0</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>()</string>
    </property>
   </widget>
   <widget class="QLabel" name="mouse_pressed">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>0</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>()</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>522</y>
      <width>90</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>490</y>
      <width>81</width>
      <height>29</height>
     </rect>
    </property>
    <property name="inputMethodHints">
     <set>Qt::InputMethodHint::ImhDigitsOnly</set>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>495</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Grid size:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_ellipse">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>490</y>
      <width>90</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Draw Ellipse</string>
    </property>
   </widget>
   <widget class="QComboBox" name="ellipse_type">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>490</y>
      <width>141</width>
      <height>28</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Polar</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Bresenham</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Cartesian</string>
     </property>
    </item>
   </widget>
   <widget class="QSpinBox" name="ellipse_a">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>490</y>
      <width>61</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="ellipse_b">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>490</y>
      <width>61</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
   </widget>
   <widget class="QLabel" name="ellipse_a_label">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>495</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>a:</string>
    </property>
   </widget>
   <widget class="QLabel" name="ellipse_b_label">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>495</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>b:</string>
    </property>
   </widget>
   <widget class="QLabel" name="polarellipsetime">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>530</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Polar: 0 ns</string>
    </property>
   </widget>
   <widget class="QLabel" name="bresenhamellipsetime">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>530</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Bresenham: 0 ns</string>
    </property>
   </widget>
   <widget class="QPushButton" name="undo">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>0</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Undo</string>
    </property>
   </widget>
   <widget class="QLabel" name="filled">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>525</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Filled: 0 pixels</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>798</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLines">
    <property name="title">
     <string>Lines</string>
    </property>
   </widget>
   <addaction name="menuLines"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>my_label</class>
   <extends>QLabel</extends>
   <header>my_label.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
